src/routing/approved.rs

```diff
@@ -1261,44 +1219,48 @@ impl Approved {
         elders_info: EldersInfo,
         proof: Proof,
     ) -> Result<Vec<Command>> {
+        let mut commands = vec![];
+
+        let prefix_is_equal = elders_info.prefix == *self.section.prefix();
+        let prefix_is_extension = elders_info.prefix.is_extension_of(self.section.prefix());
+
         let elders_info = Proven::new(elders_info, proof);
 
-        if elders_info.value.prefix == *self.section.prefix()
-            || elders_info
-                .value
-                .prefix
-                .is_extension_of(self.section.prefix())
-        {
-            self.update_barrier.handle_section_info(
-                &self.node.name(),
-                &self.section,
-                &self.network,
-                elders_info,
-            );
-            self.try_update_state()
-        } else {
-            // Other section
-            if self.network.update_neighbour_info(elders_info) {
-                self.network.prune_neighbours(self.section.prefix());
+        if prefix_is_equal || prefix_is_extension {
+            // Our section
+            if self
+                .section
+                .promote_and_demote_elders(&self.node.name())
+                .contains(&elders_info.value)
```

Comment:
As the Vote::SectionInfo could be sent from a non-elder, and the vote_accumulator only carries out aggregate,
may need to check whether the claimed elders_info is the real ones carried out the votes ?

Reply:
Do you mean to check whether a node that votes a `SectionInfo` is among the proposed elders in it? Probably a good idea.

Reply:
yes, that's what I mean. anyone can vote.

Reply:
Got it. It's addressed now.